### ISSUE TITLE
Train YOLOv8n door-state classifier + ship ONNX weights

### ISSUE THREAD

**Issue by CesMag** [Locked-Inc/STAR#453]
## What this is

Fine-tune **YOLOv8n-cls** on the **DoorDetect-Class-Dataset** to classify a rectified camera frame as `open`, `closed`, or `ajar`. Drop the resulting ONNX file into the compliance engine so the door-state classifier stops running in geometric-fallback mode.

This is the last open item from the door/obstacle compliance push. Every node that needs the classifier already works without it via the fallback path; this issue delivers the production weights.

## Why bsikar

You've got GPU access, and this is ~30 min of wall-clock time on an RTX 3060 (or ~2 hr on CPU). Full kit is committed under `scripts/door_state_training/` so you can run it with one command.

## Acceptance criteria

- [ ] ONNX weights committed at `final_capstone/compliance-engine/star_compliance/models/door_state_yolov8n.onnx`
- [ ] Test-split top-1 accuracy >= 0.80 overall
- [ ] Per-class top-1 accuracy: **open >= 0.85**, **closed >= 0.85**, **ajar >= 0.70**
- [ ] `models/README.md` updated with provenance (training date, dataset commit, hyperparameters, final accuracy, md5 of the ONNX file)
- [ ] `pytest tests/` in `compliance-engine/` passes the same 62/17 split it does today
- [ ] Manual verification via:
  ```python
  from star_compliance.detectors.door_state_classifier import DoorStateClassifier
  cls = DoorStateClassifier(weights_path='star_compliance/models/door_state_yolov8n.onnx')
  assert cls.is_available()
  ```

## Quickstart (the happy path - should take ~30 min end-to-end)

```bash
git pull origin main
bash scripts/door_state_training/train_door_state_yolov8n.sh
```

When it finishes, the script prints exact git commands to commit the weights. Follow those, push, close this issue.

## Longer version / troubleshooting

Full walkthrough: **[`scripts/door_state_training/README.md`](https://github.com/Locked-Inc/STAR/blob/main/scripts/door_state_training/README.md)**

Covers:
- Environment setup from scratch (Python 3.10+ venv, deps, CUDA verification)
- Manual dataset download if the Google Drive link rotates
- Dataset-layout normalization (`semi-open -> ajar` rename)
- Training with explicit hyperparameter recipe
- Validation protocol with per-class acceptance thresholds
- ONNX export with opset 12 + onnxsim simplification
- 7-step commit checklist at the end
- Troubleshooting entries for gdown rate-limits, CUDA OOM, accuracy stalls, ONNX opset mismatches

## Dataset

- **Source:** [gasparramoa/DoorDetect-Class-Dataset](https://github.com/gasparramoa/DoorDetect-Class-Dataset)
- **Size:** ~1,200 images, 3 classes (open, closed, semi-open -> renamed to ajar)
- **Split:** 1,086 train / 60 val / 60 test
- **Distribution:** Google Drive folder linked from the repo README

Alternatives listed at the bottom of the training README in case of license changes.

## Environment overrides

Every knob is a shell variable. Typical overrides:

```bash
# CPU-only fallback (slower but works anywhere)
DEVICE=cpu bash scripts/door_state_training/train_door_state_yolov8n.sh

# Bigger input for better ajar detection
IMG_SIZE=416 EPOCHS=75 bash scripts/door_state_training/train_door_state_yolov8n.sh

# Lower batch for small GPUs
BATCH_SIZE=16 bash scripts/door_state_training/train_door_state_yolov8n.sh
```

## Where the weights are consumed

Once the ONNX is in place, `compliance-engine/star_compliance/detectors/door_state_classifier.py` picks it up automatically via `cls.is_available()` and routes every image through OpenCV DNN -> ONNX inference. No code change needed on our side.

The runtime fallback (geometric heuristic on stereo point cloud) stays in the code as a graceful-degradation path if the ONNX file is ever missing on a deployment.

## Time budget

| Phase | RTX 3060 | CPU (8 cores) |
|---|---|---|
| Dep install | 2 min | 2 min |
| Dataset download | 5 min | 5 min |
| Training (50 epochs, 320x320) | 18 min | 110 min |
| Test split eval | 30 s | 3 min |
| ONNX export | 20 s | 30 s |
| Commit + push | 1 min | 1 min |
| **Total** | **~27 min** | **~2.1 hr** |

## After you ship

- Close this issue with a link to the commit hash
- Open a follow-up to re-train periodically as the dataset grows, or to experiment with larger backbones (yolov8s-cls) if CPU budget on the Pi 5 allows it

---

**Files you'll touch:**
- `final_capstone/compliance-engine/star_compliance/models/door_state_yolov8n.onnx` (new; the weights)
- `final_capstone/compliance-engine/star_compliance/models/README.md` (update provenance)

**Files you should NOT touch** (already set up for you):
- `scripts/door_state_training/train_door_state_yolov8n.sh`
- `scripts/door_state_training/README.md`
- `compliance-engine/star_compliance/detectors/door_state_classifier.py` (consumes the ONNX via OpenCV DNN; no changes required)

Ping `@CesMag` in a review if the acceptance thresholds come up short after a first training run - we have two fallback recipes documented (img_size=416 + 75 epochs, or switch to DeepDoors2 dataset).
